cert.go

```diff
@@ -217,7 +225,15 @@ func (opts *CertificateOptions) PutCertRequestFromMemory(wd depot.Depot) error {
 		return errors.Wrap(err, "problem getting formatted name")
 	}
 
-	if depot.CheckCertificateSigningRequest(wd, formattedName) || depot.CheckPrivateKey(wd, formattedName) {
+	csrExists, err := CheckCertificateSigningRequestWithError(wd, formattedName)
+	if err != nil {
+		return err
+	}
+	privKeyExists, err := CheckPrivateKeyWithError(wd, formattedName)
+	if err != nil {
+		return err
+	}
+	if csrExists || privKeyExists {
 		return errors.New("certificate request has existed")
```

Comment:
The error message should probably say something along the lines of "certificate request or private key already exists".